flytekit/remote/workflow.py

```diff
@@ -106,13 +107,15 @@ def promote_from_model(
         cls,
         base_model: _workflow_models.WorkflowTemplate,
         sub_workflows: Optional[Dict[_identifier.Identifier, _workflow_models.WorkflowTemplate]] = None,
+        node_launch_plans: Optional[Dict[_identifier.Identifier, _launch_plan_models.LaunchPlanSpec]] = None,
         tasks: Optional[Dict[_identifier.Identifier, _task_models.TaskTemplate]] = None,
     ) -> "FlyteWorkflow":
         base_model_non_system_nodes = cls.get_non_system_nodes(base_model.nodes)
         sub_workflows = sub_workflows or {}
         tasks = tasks or {}
         node_map = {
-            n.id: _nodes.FlyteNode.promote_from_model(n, sub_workflows, tasks) for n in base_model_non_system_nodes
+            node.id: _nodes.FlyteNode.promote_from_model(node, sub_workflows, node_launch_plans, tasks)
+            for node in base_model_non_system_nodes
```

Comment:
comment for line 123/126... it's too far away so i can't comment directly on it.  does `current << upstream_node` actually work?  `core.Node` only implements rshift not lshift and `FlyteNode` implements neither it seems.  Can we just manually append to `_upstream` to keep things simple?